engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -520,7 +525,9 @@ Object doPolyglot(
         warningProfiles[i].enter();
         anyWarnings = true;
         try {
-          accumulatedWarnings = accumulatedWarnings.append(warnings.getWarnings(r, false));
+          EnsoHashMap rWarnsMap = warnings.getWarnings(r, false);
+          Warning[] rWarnsArray = Warning.fromMapToArray(rWarnsMap, lengthNode, atNode);
+          accumulatedWarnings = accumulatedWarnings.append(rWarnsArray);
```

Comment:
A conversion to `Warning[]` remains. Not sure why? If we can obtain a `EnsoHashMap` for two (or more) values, then we can always _add them together_ as `EnsoHashMap`, can't we?

Reply:
Removed in 01381e6cc223801e389f37d6bd65ede531475e3d